lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
+    # Translate
+    #   VALUE args[6], failed, a1, a2, a3, a4, a5, a6;
+    #  into
+    #   VALUE failed, a1, a2, a3, a4, a5, a6; VALUE *args = truffle_managed_malloc(6 * sizeof(VALUE));
+
+    simple = []
+    arrays = []
+
+    line = $1
+
+    $2.split(', ').each do |local|
+      local.strip!
+      if local.end_with?(']')
+        local =~ /(\w+)\s*\[\s*(\d+)\s*\]/
+        arrays.push [$1, $2.to_i]
```

Comment:
FWIW, counting by number of files including such a pattern over Rails:
```bash
$ ack -l  '<<[^-](?!self)' **/*.rb | wc -l
418
$ ack  -l '\.push\b' **/*.rb| wc -l
43
```
`<<` seems a very clear win.